.github/workflows/cluster_endtoend.yml

```diff
@@ -28,6 +28,7 @@ jobs:
         go mod download
 
     - name: sharded cluster_endtoend
+      timeout-minutes: 30
```

Comment:
Something puzzling in this file:
```
      matrix:
        name: [11, 12, 13, 14, 15, 16, 17]
```
The matrix is supposed to go up to 25. Maybe you need to rebase?

Reply:
CI ran up to 24, so I'm assuming that the merge will be correct.